recipes-app/node-red/node-red-package.inc

```diff
@@ -12,4 +12,4 @@
 DEBIAN_DEPENDS .= ", node-red"
 RDEPENDS += "node-red"
 # install directory of nod-red-* modules
-NPM_LOCAL_INSTALL_DIR = "/root/.node-red"
+NPM_LOCAL_INSTALL_DIR = "/usr/lib"
```

Comment:
Same question as above - why not using `NODE_RED_HOME_DIR` as before?

Reply:
This is all about persistency. The preinstalled modules are installed on the RO rootfs (`/usr/lib/`), while the user installed ones are installed in `/var/lib/nodered/.node-red`, which is `$HOME/.node-red`.

Reply:
But `/var/lib` is not writable in our setup as well, is it?

Reply:
It is, The `/var` partition is persistent: https://github.com/siemens/meta-iot2050/blob/master/wic/iot2050-swu.wks.in#L17

Reply:
Ah, forgot about that.

Then the non-modifiable nodes should rather go into the location for Node-RED itself: `/usr/lib/node_modules/node-red/node_modules/`. I do not recall how to install built-in nodes there properly, but I do recall that the dashboard used to live there in older Node-RED releases, when it was still bundled.

Reply:
Wait! We install node-red-gpio like this already as it was not upstreamed at that time. Maybe just flip the destination folder in swupdate mode?

Reply:
> Wait! We install node-red-gpio like this already as it was not upstreamed at that time. Maybe just flip the destination folder in swupdate mode?

Done. It can be switched by setting `NODE_RED_IMMUTABLE = "1"`

Reply:
I did some tests with @jan-kiszka and we concluded that this logic is no longer required. We can simply install the modules in `/usr/lib` and updates via the palette are still working (they are handled locally in the `userDir`). For details, have a look at 2476f8318837caa23624c11a3f7cd7605fe61e1d